integration_tests/backend/test_backend.py

```diff
@@ -0,0 +1,76 @@
+import os
+import subprocess
+import sys
+from pathlib import Path
+
+import pytest
+import requests
+
+import aqueduct
+
+
+class TestBackend:
+    GET_WORKFLOW_TABLES_TEMPLATE = "/api/workflow/%s/tables"
+    WORKFLOW_PATH = Path(__file__).parent / "setup"
+
+    @classmethod
+    def setup_class(cls):
+        cls.client = aqueduct.Client(pytest.api_key, pytest.server_address)
+        cls.flows = {}
+
+        workflow_files = [
+            f
+            for f in os.listdir(cls.WORKFLOW_PATH)
+            if os.path.isfile(os.path.join(cls.WORKFLOW_PATH, f))
+        ]
+        for workflow in workflow_files:
+            proc = subprocess.Popen(
+                [
+                    "python3",
+                    os.path.join(cls.WORKFLOW_PATH, workflow),
+                    pytest.api_key,
+                    pytest.server_address,
+                ],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            out, err = proc.communicate()
+            out = out.decode("utf-8")
+            err = err.decode("utf-8")
+            if err:
+                raise Exception(f"Could not run workflow {workflow}.\n\n{err}")
+            else:
+                cls.flows[workflow] = out.strip().split()[-1]
+
+    @classmethod
+    def teardown_class(cls):
+        for flow in cls.flows:
+            cls.client.delete_flow(cls.flows[flow])
+
+    @classmethod
+    def get_response_class(cls, endpoint, additional_headers={}):
+        headers = {"api-key": pytest.api_key}
+        headers.update(additional_headers)
+        url = cls.client._api_client._construct_full_url(endpoint)
```

Comment:
oh interesting... this works for now, but it might be nice to decouple the full_url implementation and just use the `base_url`, since there will be endpoints we want to test in this suite that aren't available in api_client.

Reply:
`_construct_full_url` just returns `"%s%s" % (self._construct_base_url(use_https), route_suffix)` so I think re-implementing it would be duplicative,

Reply:
oh right its just the URL construction. Ok sounds good to me - maybe we can get rid of the leading underscore then?